## CI failure → fix mapping (commit `a85ea54`)

The `Scorecard action smoke — TypeScript (analysis-only)` job failed on `404cca2` with:

```
##[error]/home/runner/work/crap-rs/crap-rs/./.github/actions/scorecard/action.yml (Line: 242, Col: 18):
  Unrecognized named-value: 'runner'. Located at position 1 within expression: runner.os
##[error]Failed to load /home/runner/work/crap-rs/crap-rs/./.github/actions/scorecard/action.yml
```

### Root cause

The new `html-artifact-name-suffix` input's `description: |` block-scalar contained the literal string `-${{ runner.os }}` as illustrative syntax explaining what the default expands to. GitHub's composite-action manifest loader **evaluates `${{ ... }}` expressions inside `description:` text** (not just inside `value:` / `with:` / `run:` contexts), so `runner.os` was resolved at action-LOAD time — before any step context exists — and the loader rejected the manifest.

The actual functional use of `runner.os` at `steps.upload-rust.with.name:` (and `steps.upload-typescript.with.name:`) is **correct** — `runner.os` is in scope inside step `with:` blocks. The bug was scoped strictly to documentation prose.

### Fix

Rephrase the description text + a nearby comment to reference `<runner.os>` (literal angle-bracketed placeholder, common docs convention) instead of GHA expression syntax. Same documentation intent, no expression evaluation at load time. Two spots:

| Location | Before | After |
|---|---|---|
| `inputs.html-artifact-name-suffix.description` | `defaults to \`-${{ runner.os }}\`` | `defaults to \`-<runner.os>\` (Linux/macOS/Windows)` |
| Comment above `Upload Rust HTML report` step | `the default \`-${{ runner.os }}\`` | `the default \`-<runner.os>\`` |

### Verification

- `grep -nE '\$\{\{' action.yml` inside inputs/outputs blocks now matches only `value:` fields (the only valid expression context at action manifest top level)
- Behavior unchanged: `steps.upload-{rust,typescript}.with.name:` still resolves `runner.os` correctly at step-runtime

### Lesson PARKed

Add to crap-rs Composite-action gotchas log: **`description:` block-scalars of `inputs:` are evaluated as GHA expressions, not treated as opaque docs**. Anyone documenting GHA syntax inside an input description must escape `${{` (or use angle-bracket placeholder convention like the fix). README.md is unaffected (rendered by GitHub.com as markdown, not parsed by the action manifest loader).

— orchestrator session (parent Claude, model Opus 4.7)

--- a/.github/actions/scorecard/action.yml
+++ b/.github/actions/scorecard/action.yml
@@ -1338,7 +1338,11 @@ runs:
       if: steps.lang.outputs.language == 'rust' || steps.lang.outputs.language == 'typescript' || steps.lang.outputs.language == 'multi'
       shell: bash
       env:
-        MARKDOWN: ${{ steps.run.outputs.markdown }}
+        # Markdown body is read FROM the temp files Run analysis already
+        # wrote (rather than re-piped through an env var) so we don't
+        # tip the 64 KB env-var limit on large projects — gemini catch
+        # on PR #312. Filenames: `${BIN}-scorecard.md` for single-lang,
+        # `scorecard-combined.md` for multi-lang (see L898 + L1050).
         HTML_REPORT: ${{ inputs.html-report }}
         IS_MULTI: ${{ steps.presets.outputs.is_multi }}
         LANGUAGE: ${{ steps.lang.outputs.language }}
@@ -1347,8 +1351,13 @@ runs:
       run: |
         set -euo pipefail
         composed_file="$RUNNER_TEMP/scorecard-sticky-message.md"
-        : > "$composed_file"
-        printf '%s' "$MARKDOWN" > "$composed_file"
+        if [ "$LANGUAGE" = "multi" ]; then
+          cat "$RUNNER_TEMP/scorecard-combined.md" > "$composed_file"
+        elif [ "$LANGUAGE" = "rust" ]; then
+          cat "$RUNNER_TEMP/crap4rs-scorecard.md" > "$composed_file"
+        else
+          cat "$RUNNER_TEMP/crap4ts-scorecard.md" > "$composed_file"
+        fi
 
         if [ "$HTML_REPORT" = "true" ]; then
           # Guard the link append on actual upload success — when
@@ -1357,19 +1366,23 @@ runs:
           # the action-level outputs are documented as empty in that
           # case. Append the link only for languages that produced a
           # URL.
+          #
+          # URLs wrapped in `<...>` per the GFM convention — keeps the
+          # link renderable even if a future artifact-URL shape includes
+          # parentheses or spaces (gemini catch on PR #312).
           link_block=""
           if [ "$IS_MULTI" = "true" ]; then
             if [ -n "$HTML_URL_RUST" ] && [ -n "$HTML_URL_TS" ]; then
-              link_block="📊 [Download full HTML report — Rust]($HTML_URL_RUST) · [TypeScript]($HTML_URL_TS)"
+              link_block="📊 [Download full HTML report — Rust](<$HTML_URL_RUST>) · [TypeScript](<$HTML_URL_TS>)"
             elif [ -n "$HTML_URL_RUST" ]; then
-              link_block="📊 [Download full HTML report — Rust]($HTML_URL_RUST)"
+              link_block="📊 [Download full HTML report — Rust](<$HTML_URL_RUST>)"
             elif [ -n "$HTML_URL_TS" ]; then
-              link_block="📊 [Download full HTML report — TypeScript]($HTML_URL_TS)"
+              link_block="📊 [Download full HTML report — TypeScript](<$HTML_URL_TS>)"
             fi
           elif [ "$LANGUAGE" = "rust" ] && [ -n "$HTML_URL_RUST" ]; then
-            link_block="📊 [Download full HTML report]($HTML_URL_RUST)"
+            link_block="📊 [Download full HTML report](<$HTML_URL_RUST>)"
           elif [ "$LANGUAGE" = "typescript" ] && [ -n "$HTML_URL_TS" ]; then
-            link_block="📊 [Download full HTML report]($HTML_URL_TS)"
+            link_block="📊 [Download full HTML report](<$HTML_URL_TS>)"
           fi
           if [ -n "$link_block" ]; then
             # Two trailing newlines BEFORE the link block guarantee a

